### PR TITLE
version-guard default move constructors for MSVC

### PR DESCRIPTION
**Pull request by nschloe**
Follow-up on PR #1100.

### CODE REVIEW

File: src/util/audiosignal.h [hunk 1]
```diff
@@ -75,8 +75,8 @@ class AudioSignal {
     AudioSignal(const AudioSignal&) = default;
```
Comment (uklotzde) [inline src/util/audiosignal.h]: I didn't notice: Why have you added all those defaulted constructors and assignment operators? They are not needed. We only need to declare a virtual destructor with a default implementation.

Comment (nschloe) [inline src/util/audiosignal.h]: Took it from [here](http://stackoverflow.com/a/4782927/353337):

> Note that move constructor and move assignment operator won't be generated for a class that explicitly declares any of the other special member functions, that copy constructor and copy assignment operator won't be generated for a class that explicitly declares a move constructor or move assignment operator, [...]

**Comment by uklotzde** [comment]
Corect, but we neither need nor want them. Declare only the defaulted 
destructor and everything is fine.

The same might apply to the other classes. Don't add assignment operators, 
unless you need them.

**Comment by nschloe** [comment]
Done.

**Comment by nschloe** [comment]
Turns out we need most of the constructors. Removed those that we don't absolutely need (there are two of them).

**Comment by uklotzde** [comment]
The semantics of move assignment are complicated and it is almost never 
needed. I don't see any use case here.

**Comment by nschloe** [comment]
> I don't see any use case here.

They are really hard to see, too, and the code compiles and runs fine without them. Just slower here and there. That's why some argue to always create move constructors so they're there and silently do their job when needed.

> The semantics of move assignment are complicated

It's two lines of code for four classes, some would argue it's not too messy. A cleaner solution, as mentioned before, would be to remove all of the `=default` stuff and have it automatically generated by the compiler.

**Comment by daschuer** [comment]
LGTM, QString and QImage are movable in QT 5 so there will be a benefit from move constructors. 
Thank you.

**Pull request merged**